use-sync-external-store.native.js

```diff
@@ -0,0 +1 @@
+export { useSyncExternalStore } from "use-sync-external-store/shim/index.native.js";
```

Comment:
I'm not familiar with how modules are bundled in native, is this some kind naming convention? I see that this file isn't actually referenced anywhere else.

Reply:
@molefrog it is referenced to the same degree that the `.js` version is, since we are not specifying any extension at all when we import it. It is up to the bundling system to select which extension to use, whether that be `.js` or `.native.js`!

That being said, I'm not too familiar with react native either, all I know is that I copied this fix over verbatim from react-query and it worked for them. 😄